server/server.js

```diff
@@ -1,13 +1,32 @@
 const path = require('path');
 const express = require('express');
+const tree = require('./tree');
 
 const app = express();
 
-app.use(express.static(path.resolve(__dirname, '../client')));
+app.use((req, res, next) => {
+  res.header('Access-Control-Allow-Origin', '*');
+  res.header('Access-Control-Allow-Headers', 'Origin, X-Requested-With, Content-Type, Accept');
+  next();
+});
+
+app.use(express.static(path.resolve(__dirname, '../client/public')));
+
 app.listen(process.env.npm_package_config_port);
 
-app.get('/', (req, res) => {
-  res.send('Connected');
+app.get('/api/tree', (req, res) => {
+  if (!Object.keys(req.query).length 
```

Comment:
I would ditch this check. We just care about `!req.query.path || !req.query.path.trim().length` (in this order so you don't call `.trim()` on undefined/null;